Visual Shaders: Make output-ports for vector types expandable by default

DennisManaa opened the pull request:
This PR is a continuation of the work done in: #73041

# Changes made

On the left side you can see the state before the changes I made (which is the same as in #73041) and on the right side you can see the state after the changes I made. The following changes were made:

* **Marked in red**: Removed the expand buttons from the vector-components.
* **Marked in blue**: Renamed the component-ports, to better reflect which vector-component is meant (as suggested by @Norrox).
* **Marked in yellow**: Reversed the directions of the collapse/expand-arrows when open/closed, which aligns them with other occurrences of these buttons.
* **Marked in magenta**: Fixed outputs of `Vector2Parameter`, `Vector3Parameter`, `Vector4Parameter` and `CurveXYZTexture`-Nodes.
* **Marked in cyan**: Fixed many nodes that weren't working correctly (the preview and output of vector-components were interpreted as if they were vector-outputs, what caused ports of several nodes to behave strangely)

![pr_marked_changes_01](https://github.com/godotengine/godot/assets/26871131/a2a5f498-cea1-4cbc-af49-e8af54454bb4)
![pr_marked_changes_02](https://github.com/godotengine/godot/assets/26871131/e5a33c55-ea51-4e62-98ae-7a2eaf514cbb)
![pr_marked_changes_03](https://github.com/godotengine/godot/assets/26871131/41dc5c1b-57d0-466e-ae27-f5208e03beee)
![pr_marked_changes_04](https://github.com/godotengine/godot/assets/26871131/b5846800-dd99-4506-b7ce-009b3dc813bf)
![pr_marked_changes_05](https://github.com/godotengine/godot/assets/26871131/c11de2ac-d2c0-4105-b5e0-e0150d6ce407)

DennisManaa committed:
made visual shader ports expandable by default if there is only one output port and it's of any vector type

Co-authored-by: QbieShay <cislaghi.ilaria@gmail.com>

Chaosus commented:
Thanks, this PR looks very good. Please squash commits as required by our pipeline (see https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#making-changes)

QbieShay commented:
PR looks good to me, my only comment on this is that I find the red(x) syntax very hard to read. I'd rather we remove this change and discuss a bit wheter we want to have some nodes be rgba and some nodes directly xyzw, or find some less visually cluttered ways to represent this.

QbieShay reviewed (changes requested):
PR looks good to me, my only comment on this is that I find the red(x) syntax very hard to read. I'd rather we remove this change and discuss a bit wheter we want to have some nodes be rgba and some nodes directly xyzw, or find some less visually cluttered ways to represent this.

Norrox commented:
> PR looks good to me, my only comment on this is that I find the red(x) syntax very hard to read. I'd rather we remove this change and discuss a bit wheter we want to have some nodes be rgba and some nodes directly xyzw, or find some less visually cluttered ways to represent this.

I like this PR 😁, would it be better to remove red green blue text completely and for example just write X in color red and the port be red as it is?

AThousandShips commented:
> example just write X in color red and the port be red as it is

That's not appropriate for accessibility

QbieShay commented:
I'd rather get this pr merged soon and discuss on those UI changes separately, possibly in a proposal

DennisManaa commented:
Thank you very much for your rapid feedback. I'm glad to hear that you like this PR :)
I squashed the commits and reverted the changes regarding the naming of the ports.

DennisManaa commented:
Sorry for this commit...

I just realized I made a mistake in the commit message. The 'Co-authored-by:' tag wasn't recognized correctly, so I've corrected it. I built on top of QbieShay's work, so it should correctly be represented in the commit! :)

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:
